### ISSUE TITLE
understanding validation part

### ISSUE THREAD

**Issue by nealzGu** [fizyr/keras-retinanet#294]
Hi guys.
If you have a minute, please help me understand this code a little bit more.

the loss that gets printed to terminal during training (l1 for regression, focal for classification). Is that for the training set or validation set? 

How is the validation_generator used during training? (like when and how often do the validation happens, etc.)

I can't find any hint of that in the code, or maybe I just don't see it. Thanks for pointing me into the right direction or code snippet :)

**Comment by hgaiser** [comment]
The losses printed are for the training generator, the validation generator is only used in the evaluation callback, which computes the mAP score. This happens every epoch. 

https://github.com/fizyr/keras-retinanet/blob/master/keras_retinanet/bin/train.py#L131

**Comment by nealzGu** [comment]
thanks for the reply!

I noticed that this part is not in the current version of my "train.py", but I will download the new versions and try it out.

Can I print out the mAP values in terminal during training? or can I only see them when looking at tensorboard?

**Comment by hgaiser** [comment]
Tensorboard is only additional, by default it will print the mAP in the terminal.

**Comment by nealzGu** [comment]
it's working with the new version. thanks a lot!
